src/lib/battery/battery.cpp

```diff
@@ -226,6 +226,12 @@ void Battery::estimateRemaining(const float voltage_v, const float current_a, co
 			// directly apply current capacity slope calculated using current
 			_remaining -= _discharged_mah_loop / _params.capacity;
 			_remaining = math::max(_remaining, 0.f);
+
+			// calculation should never exceed percentage based on raw current consumption
+			_remaining_current = 1 - _discharged_mah / _params.capacity;
```

Comment:
Put all of this within a check that `_params.capacity` (BATn_CAPACITY) is > 0?

Reply:
Its already within that if statement

Reply:
https://github.com/PX4/PX4-Autopilot/blob/73ab87e4f127ada15f99ef4ef90c85fc3ec01ad1/src/lib/battery/battery.cpp#L216